hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -232,6 +234,33 @@ public Token loadToken(Id id) {
 		return token;
 	}
 
+	/**
+	 * Returns a {@link UniqueToken} model of the requested unique token, with operations that can be used to
+	 * implement business logic in a transaction.
+	 *
+	 * @param token the token model, on which to load the of the unique token
+	 * @param serialNumbers the serial numbers to load
+	 * @throws InvalidTransactionException if the requested token class is missing, deleted, or expired and pending removal
+	 */
+	public void loadUniqueTokens(Token token, List<Long> serialNumbers) {
+		var tokenId = token.getId();
+		final var tokenKey = new MerkleEntityId(tokenId.getShard(), tokenId.getRealm(), tokenId.getNum());
+		final var merkleToken = tokens.get().get(tokenKey);
+
+		validateUsable(merkleToken);
```

Comment:
I am not sure why we need to get the merkle token again since we already did that (we are providing the `Token token` after all. Maybe your idea was the `loadUniqueTokens` should be validating the token and be resilient, however I am not sure that the additional logic is worth it here